### PR TITLE
New index injection

### PR DESCRIPTION
**Pull request by Mironor**

### CODE REVIEW

File: batch/src/main/scala/uk/gov/ons/addressindex/utils/SparkProvider.scala [hunk 1]
```diff
@@ -37,11 +37,53 @@ object SparkProvider {
...
+    val paoLeftRangeExists = paoStartNumber.nonEmpty || paoStartSuffix.nonEmpty
+    val paoRightRangeExists = paoEndNumber.nonEmpty || paoEndSuffix.nonEmpty
+    val paoHyphen = if (paoLeftRangeExists && paoRightRangeExists) "-" else ""
+
+    val paoNumbers = Seq(paoStartNumber, paoStartSuffix, paoHyphen, paoEndNumber, paoEndSuffix)
+      .map(_.trim).mkString
+    val pao =
+      if (paoText == organisation || paoText.isEmpty) paoNumbers
+      else if (paoNumbers.isEmpty) s"$paoText"
+      else s"$paoText $paoNumbers"
```
Comment (saniemi) [inline batch/src/main/scala/uk/gov/ons/addressindex/utils/SparkProvider.scala]: Is is correct that paoText comes before paoNumbers? I would have assumed that they were the other way around e.g. "65B Victorian House".

File: batch/src/main/scala/uk/gov/ons/addressindex/utils/SqlHelper.scala [hunk 2]
```diff
@@ -65,19 +95,37 @@ object SqlHelper {
...
+
+        val lpiPostCode: Option[String] = lpis.headOption.flatMap(_.get("postcodeLocator").map(_.toString))
+        val pafPostCode: Option[String] = pafs.headOption.flatMap(_.get("postcode").map(_.toString))
+
+        val postCode = if (pafPostCode.isDefined) pafPostCode.getOrElse("")
+        else lpiPostCode.getOrElse("")
+
+        val splitPostCode = postCode.split(" ")
+        val (postCodeOut, postCodeIn) =
+          if (splitPostCode.size == 2 && splitPostCode(1).length == 3) (splitPostCode(0), splitPostCode(1))
```
Comment (saniemi) [inline batch/src/main/scala/uk/gov/ons/addressindex/utils/SqlHelper.scala]: Good check!

File: batch/src/main/resources/es_index.json [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+            "source": {
+              "type": "string",
+              "index": "not_analyzed"
+            },
+            "nagAll": {
+              "type": "string",
+              "analyzer": "welsh_split_analyzer"
+            },
+            "relatives": {
+              "type": "long",
```
Comment (Mironor) [inline batch/src/main/resources/es_index.json]: @Ngwara has a structure to store relatives

File: batch/src/main/resources/es_index.json [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+              "analyzer": "welsh_no_split_analyzer"
+            },
+            "pafAll": {
+              "type": "string",
+              "analyzer": "welsh_split_analyzer"
+            }
+          }
+        },
+        "uprn": {
+          "type": "long",
```
Comment (Mironor) [inline batch/src/main/resources/es_index.json]: It should be the same, yes.
It's easier to have it there, we group pafs and nags by uprn that results in a hybrid address.
Removing the `uprn` column will add some code (and complexity), it's a small duplication, I think it does not hurt.

File: batch/src/main/scala/uk/gov/ons/addressindex/utils/SparkProvider.scala (hunk 1 above)
Comment (Mironor) [inline batch/src/main/scala/uk/gov/ons/addressindex/utils/SparkProvider.scala]: This was a copy from the Api project.
There should be a comma between paoText ad pao Numbers

File: batch/src/main/resources/es_index.json (hunk 4 above)
Comment (saniemi) [inline batch/src/main/resources/es_index.json]: Agreed. Given that the index is very small, I agree that this level of duplication does not really hurt.

File: batch/src/main/scala/uk/gov/ons/addressindex/utils/SparkProvider.scala (hunk 1 above)
Comment (saniemi) [inline batch/src/main/scala/uk/gov/ons/addressindex/utils/SparkProvider.scala]: Ok.

File: batch/src/main/resources/es_index.json (hunk 3 above)
Comment (Mironor) [inline batch/src/main/resources/es_index.json]: We should add `"index":"no"` to exclude it from search

**Pull request merged**